### PR TITLE
Login-modal should get it's own handling, split out from index.js

### PR DESCRIPTION
**Pull request by Coenego**
The current implementation of the login-modal isn't up to par with the rest of the modals.

It should only be rendered and shown when an anonymous user triggers an action that can only be done once signed in.

### CODE REVIEW

**Comment by Coenego** [comment]
Assigning to @bp323 for review

File: shared/gh/js/views/gh.login-modal.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        // Render the login modal template
+        gh.utils.renderTemplate('login-modal', {
+            'data': {
+                'gh': gh,
+                'isGlobalAdminUI': false
+            }
+        }, $('#gh-modal'), function() {
+
+            // Add logic once the modal is shown
+            $('#gh-modal-login').on('shown.bs.modal', function() {
```
Comment (bp323) [inline shared/gh/js/views/gh.login-modal.js]: Can this be moved to `addBinding`?

File: shared/gh/js/views/gh.login-modal.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                gh.utils.trackEvent(['Navigation', 'Authentication modal triggered']);
+
+                // Bind the validator to the login form
+                $('.modal-body .gh-signin-form').validator({
+                    'disable': false
+                }).on('submit', doLogin);
+            });
+
+            // Put the focus back on the trigger on hide
+            $('#gh-modal-login').on('hidden.bs.modal', function() {
```
Comment (bp323) [inline shared/gh/js/views/gh.login-modal.js]: Can this be moved into `addbinding` as well?

**Comment by bp323** [comment]
Looks good, only a few minor comments remain

**Comment by Coenego** [comment]
Back to @bp323

**Comment by bp323** [comment]
Verified and merged.

**Pull request merged**